src/DataValidation.py

```diff
@@ -192,6 +195,110 @@ def checkItemsThatShouldBeRequired():
+            for result in re.findall(r'\{ItemValue\(([^:]*)\:([^)]+)\)\}', requires):
+                value = result[0].lower().strip()
+                count = int(result[1])
+                if not values_requested.get(value):
+                    values_requested[value] = count
+                else:
+                    values_requested[value] = max(values_requested[value], count)
+        return values_requested
+
+    @staticmethod
+    def checkIfEnoughItemsForValue():
+        values_available = {}
+        values_requested = {}
+
+        # First find the biggest values required by locations
+        for location in DataValidation.location_table:
+            if "requires" not in location:
+                continue
+
+            # convert to json so we don't have to guess the data type
+            location_requires = json.dumps(location["requires"])
+
+            DataValidation._requireItemValueRegex(values_requested, location_requires)
+        # Second, check region requires for the presence of item name
+        for region_name in DataValidation.region_table:
+            region = DataValidation.region_table[region_name]
+
+            if "requires" not in region:
+                continue
+
+            # convert to json so we don't have to guess the data type
+            region_requires = json.dumps(region["requires"])
+
+            DataValidation._requireItemValueRegex(values_requested, region_requires)
+        # then if something is requested, we loop items
+        if values_requested:
+
+            # get all the available values with total count
+            for item in DataValidation.item_table:
+                # if the item is already progression, no need to check
+                if not item.get("progression") and not item.get("progression_skip_balancing"):
+                    continue
+
+                item_count = item.get('count', None)
+                if item_count is None: #check with none because 0 == false
+                    item_count = '1'
+
+                for key, count in item.get("value", {}).items():
+                    if not values_available.get(key.lower().strip()):
+                        values_available[key] = 0
+                    values_available[key] += int(count) * int(item_count)
+
+            # compare whats available vs requested
+            errors = []
+            for value, count in values_requested.items():
+                if values_available.get(value, 0) < count:
+                    errors.append(f"       '{value}': {values_available.get(value, 0)} out of the {count} {value} worth of progression items required can be found.")
+            if errors:
+                raise ValidationError("There are not enough progression items for the following values: \n" + "\n".join(errors))
+
+    @staticmethod
+    def preFillCheckIfEnoughItemsForValue(world: World, multiworld: MultiWorld):
+        from .Helpers import get_items_with_value, get_items_for_player

```

Comment:
if I import that at the top it cause a circular import.

Reply:
That's fine